xonsh/vox.py

```diff
@@ -0,0 +1,198 @@
+from os.path import join, basename, exists
+import os
+from shutil import rmtree
+import venv
+
+import builtins
+
+
+class Vox:
+    """Vox is a virtual environment manager for xonsh."""
+
+    def __init__(self):
+        """Ensure that $VIRTUALENV_HOME is defined and declare the available vox commands"""
+
+        if not builtins.__xonsh_env__.get('VIRTUALENV_HOME'):
+            home_path = os.path.expanduser('~')
+
+            builtins.__xonsh_env__['VIRTUALENV_HOME'] = join(home_path, '.virtualenvs')
+
+        self.commands = {
+            ('new',): self.create_env,
+            ('activate', 'workon', 'enter'): self.activate_env,
+            ('deactivate', 'exit'): self.deactivate_env,
+            ('list', 'ls'): self.list_envs,
+            ('remove', 'rm', 'delete', 'del'): self.remove_env,
+            ('help', '-h', '--help'): self.show_help
+        }
+
+    def __call__(self, args, stdin=None):
+        """Call the right handler method for a given command."""
+
+        if not args:
+            self.show_help()
+            return None
+
+        command_name, params = args[0], args[1:]
+
+        try:
+            command = [
+                self.commands[aliases] for aliases in self.commands
+                if command_name in aliases
+            ][0]
+
+            command(*params)
+
+        except IndexError:
+            print('Command "%s" doesn\'t exist.\n' % command_name)
+            self.print_commands()
+
+    @staticmethod
+    def create_env(name):
+        """Create a virtual environment in $VIRTUALENV_HOME with python3's ``venv``.
+
+        Parameters
+        ----------
+        name : str
+            Virtual environment name
+        """
+
+        env_path = join(builtins.__xonsh_env__['VIRTUALENV_HOME'], name)
+
+        print('Creating environment...')
+
+        venv.create(env_path, with_pip=True)
+
+        print('Environment "%s" created. Activate it with "vox activate %s".\n' % (name, name))
+
+    @staticmethod
+    def activate_env(name):
+        """Activate a virtual environment.
+
+        Parameters
+        ----------
+        name : str
+            Virtual environment name
+        """
+
+        env_path = join(builtins.__xonsh_env__['VIRTUALENV_HOME'], name)
+
+        if not exists(env_path):
+            print('This environment doesn\'t exist. Create it with "vox new %s".\n' % name)
+            return None
+
+        if os.name == 'nt':
+            bin_dir = 'Scripts'
+
+        elif os.name == 'posix':
```

Comment:
You should probably import `ON_WINDOWS` and `ON_POSIX` from `xonsh.tools` and use them here, for consistency.